react-search-component/my-app/src/SearchableList.jsx

```diff
@@ -0,0 +1,41 @@
+import './SearchableList.css';
+import { useState } from 'react';
+import { FaMagnifyingGlass } from 'react-icons/fa6';
+
+export default function SearchableList({ list }) {
+  const [value, setValue] = useState('');
+  let currentRender = [...list];
+
+  if (!value) {
+    currentRender = [...list];
+  } else {
+    currentRender = [];
+    for (let i = 0; i < list.length; i++) {
+      if (list[i].text.toLowerCase().includes(value.toLowerCase())) {
+        currentRender.push(list[i]);
+      }
+    }
+  }
```

Comment:
You could more easily create `currentRender` using `list.filter` plus `list.map`.